server/api/src/main/java/org/hawkular/apm/server/api/model/zipkin/Span.java

```diff
@@ -225,4 +231,27 @@ public String componentType() {
         return null;
     }
 
+    public List<Property> properties() {
+        List<Property> properties = new ArrayList<>();
+
+        if (binaryAnnotations != null) {
+            Set<String> services = new HashSet<>();
+
+            for (BinaryAnnotation binaryAnnotation : binaryAnnotations) {
+                Property property = new Property(binaryAnnotation.getKey(), binaryAnnotation.getValue());
+
+                Endpoint endpoint = binaryAnnotation.getEndpoint();
+                if (endpoint != null &&
+                        !services.contains(endpoint.getServiceName())) {
+
+                    properties.add(new Property("service", endpoint.getServiceName(), PropertyType.Text));
+                    services.add(endpoint.getServiceName());
+                }
```

Comment:
One thing I don't understand about the zipkin model is why the endpoint is contained within the annotation, so may be repeated multiple times within the same span, but with different values (i.e. service names).

This would seem to imply that the annotations could be associated with different endpoints - not sure if this is actually the case.

But just wondering if we should initially just assume only one service? and if multiple provided then generate a trace message?

Just thinking out loud, so not necessarily suggesting a change, but not sure how this should be handled if multiple services are actually provided. The properties can handle multiple values, but not sure how to deal with it at a high level (i.e. in the UI, etc).